mllib-dal/src/main/scala/com/intel/oap/mllib/stat/SummarizerDALImpl.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.oap.mllib.stat
+
+import com.intel.oap.mllib.{OneCCL, OneDAL}
+import org.apache.spark.TaskContext
+import org.apache.spark.internal.Logging
+import org.apache.spark.ml.linalg.{Vector}
+import org.apache.spark.mllib.linalg.{Vectors => OldVectors}
+import org.apache.spark.mllib.stat.{MultivariateStatisticalDALSummary, MultivariateStatisticalSummary => Summary}
+import org.apache.spark.rdd.RDD
+import com.intel.oap.mllib.Utils.getOneCCLIPPort
+
+
+class SummarizerDALImpl(
+                          val executorNum: Int,
+                          val executorCores: Int)
+  extends Serializable with Logging {
+
+  def computeSummarizerMatrix(data: RDD[Vector]): Summary = {
+    val kvsIPPort = getOneCCLIPPort(data)
+
+    val sparkContext = data.sparkContext
+    val useGPU = sparkContext.getConf.getBoolean("spark.oap.mllib.useGPU", false)
+
+    val coalescedTables = OneDAL.rddVectorToMergedTables(data, executorNum)
+
+    val results = coalescedTables.mapPartitionsWithIndex { (rank, table) =>
+
+      val gpuIndices = if (useGPU) {
+        val resources = TaskContext.get().resources()
+        resources("gpu").addresses.map(_.toInt)
+      } else {
+        null
+      }
+
+      val tableArr = table.next()
+      OneCCL.init(executorNum, rank, kvsIPPort)
+
+      val computeStartTime = System.nanoTime()
+
+      val result = new SummarizerResult()
+      cSummarizerTrainDAL(
+        tableArr,
+        executorNum,
+        executorCores,
+        useGPU,
+        gpuIndices,
+        result
+      )
+
+      val computeEndTime = System.nanoTime()
+
+      val durationCompute = (computeEndTime - computeStartTime).toDouble / 1E9
+
+      logInfo(s"SummarizerDAL compute took ${durationCompute} secs")
+
+      val ret = if (OneCCL.isRoot()) {
+
+        val convResultStartTime = System.nanoTime()
+        val meanMatrix = OneDAL.numericTableToOldMatrix(OneDAL.makeNumericTable(result.meanNumericTable))
```

Comment:
The code is not decent here to convert twice. 
There is a logical problem in the function numericTableToOldMatrix. isTransposed should be set to true. This function is not used, you can remove it. 
Is result.meanNumericTable a Nx1 or 1xN matrix? For Nx1, You can call numericTableNx1ToVector, For 1xN, you can modify numericTableNx1ToVector function to numericTable1xNToVector.